etc/docker/elyra/Dockerfile.dev

```diff
@@ -15,28 +15,70 @@
 # limitations under the License.
 #
 
-# Ubuntu 18.04 LTS - Bionic
+# Ubuntu 20.04.2 LTS (Focal Fossa)
 # Repository: https://hub.docker.com/r/jupyterhub/k8s-singleuser-sample/tags
 FROM jupyterhub/k8s-singleuser-sample:1.2.0
 
-
 ARG TAG="dev"
 
 USER root
 
-ADD  start-elyra.sh /usr/local/bin/start-elyra.sh
-COPY elyra /tmp/elyra
+# Install build dependencies
+RUN apt-get -yq update \
+ && apt-get -yq install --no-install-recommends \
+    build-essential \
+    curl \
+    git \
+    sudo \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+# Install NodeJS
+RUN curl -sL https://deb.nodesource.com/setup_12.x | bash - \
+ && apt-get -yq update \
+ && apt-get -yq install --no-install-recommends \
+    nodejs \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+# Install Yarn
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
+ && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
+ && apt-get -yq update \
+ && apt-get -yq install --no-install-recommends \
+    yarn \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+USER ${NB_USER}
+
+# Copy Elyra source code
+COPY --chown=${NB_USER} --chmod=0755 elyra /tmp/elyra/
 
-RUN chmod ugo+x /usr/local/bin/start-elyra.sh && \
-    fix-permissions /tmp/elyra && \
-    apt-get update && apt-get install -y build-essential && \
-    rm -rf /var/lib/apt/lists/*
+# Install Elyra from source
+RUN cd /tmp/elyra \
+ && make install \
```

Comment:
Should we use the new `install-all` `Makefile` target instead? This would better align with the build sequence in `Dockerfile` (`pip install ... elyra[all]`) and eliminate the need to perform the `pip install ... elyra-examples-*` below

Reply:
Done in https://github.com/elyra-ai/elyra/pull/2344/commits/c5e17124f04ab5d2016ceb558cda1281e146ca12